src/components/subscriptions/data/hooks.js

```diff
@@ -0,0 +1,159 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import LicenseManagerApiService from './service';
+import NewRelicService from '../../../data/services/NewRelicService';
+import {
+  licenseStatusByTab,
+  NETWORK_ERROR_MESSAGE,
+  SUBSCRIPTION_USERS,
+  SUBSCRIPTION_USERS_OVERVIEW,
+  SUBSCRIPTIONS,
+} from './constants';
+import { camelCaseObject } from '../../../utils';
+
+/*
+This hook provides all subscription data for the authenticated user and given enterprise customer UUID.
+ */
+export const useSubscriptions = ({ enterpriseId, errors, setErrors }) => {
+  const [subscriptions, setSubscriptions] = useState({
+    results: [],
+    count: 0,
+    next: null,
+    previous: null,
+  });
+
+  const forceRefresh = () => {
+    setSubscriptions({ ...subscriptions });
+  };
+
+  useMemo((page = 1) => {
```

Comment:
will `page` ever be set/ changed by anyone? if so, where would that be?

Reply:
I don't believe it will. Page number was part this hook beforehand and has always been used as a query param in the initial call to license-manager so I left it in there. I agree that it doesn't seem needed though

Reply:
Just to confirm, does the API get called with the appropriate page number during pagination of the TabContentTable data?

Reply:
I confirmed that it does